Hydrogen Installation: Uncaught Error: Cannot find module 'react-hot-loader'

[Enter steps to reproduce:]

1. Install Atom 64 Bit on Windows 7 Enterprise
2. Go to Atom package installer, search for and select Hydrogen, press install.

**Atom**: 1.27.2 x64
**Electron**: 1.7.15
**OS**: Microsoft Windows 7 Enterprise 
**Thrown From**: [Hydrogen](ssh://git@github.com/nteract/hydrogen) package 2.5.0


### Stack Trace

Uncaught Error: Cannot find module 'react-hot-loader'

```
At module.js:472

Error: Cannot find module 'react-hot-loader'
    at Module._resolveFilename (module.js:470:15)
    at Module._resolveFilename (~/AppData/Local/atom/app-1.27.2/resources/electron.asar/common/reset-search-paths.js:35:12)
    at Function.get_Module._resolveFilename (~/AppData/Local/atom/app-1.27.2/resources/app/src/module-cache.js:354:58)
    at Module.require (/app.asar/static/index.js:40:43)
    at require (~/AppData/Local/atom/app-1.27.2/resources/app/src/native-compile-cache.js:66:33)
    at /packages/Hydrogen/node_modules/@nteract/transform-vdom/lib/object-to-react.js:13:21
    at /packages/Hydrogen/node_modules/@nteract/transform-vdom/lib/object-to-react.js:16:3)
    at /packages/Hydrogen/node_modules/@nteract/transform-vdom/lib/object-to-react.js:163:3)
    at Module.get_Module._compile (~/AppData/Local/atom/app-1.27.2/resources/app/src/native-compile-cache.js:106:36)
    at Object.value [as .js] (~/AppData/Local/atom/app-1.27.2/resources/app/src/compile-cache.js:240:29)
    at Module.load (module.js:488:32)
    at tryModuleLoad (module.js:447:12)
    at Function.Module._load (module.js:439:3)
    at Module.require (/app.asar/static/index.js:47:45)
    at require (~/AppData/Local/atom/app-1.27.2/resources/app/src/native-compile-cache.js:66:33)
    at /packages/Hydrogen/node_modules/@nteract/transform-vdom/lib/index.js:14:22)
    at /packages/Hydrogen/node_modules/@nteract/transform-vdom/lib/index.js:114:3)
    at Module.get_Module._compile (~/AppData/Local/atom/app-1.27.2/resources/app/src/native-compile-cache.js:106:36)
    at Object.value [as .js] (~/AppData/Local/atom/app-1.27.2/resources/app/src/compile-cache.js:240:29)
    at Module.load (module.js:488:32)
    at tryModuleLoad (module.js:447:12)
    at Function.Module._load (module.js:439:3)
    at Module.require (/app.asar/static/index.js:47:45)
    at require (~/AppData/Local/atom/app-1.27.2/resources/app/src/native-compile-cache.js:66:33)
    at /packages/Hydrogen/node_modules/@nteract/transforms/lib/index.js:43:22)
    at /packages/Hydrogen/node_modules/@nteract/transforms/lib/index.js:110:3)
    at Module.get_Module._compile (~/AppData/Local/atom/app-1.27.2/resources/app/src/native-compile-cache.js:106:36)
    at Object.value [as .js] (~/AppData/Local/atom/app-1.27.2/resources/app/src/compile-cache.js:240:29)
    at Module.load (module.js:488:32)
    at tryModuleLoad (module.js:447:12)
```

### Commands

```
     -5:05.0 core:confirm (input.hidden-input)
```

### Non-Core Packages

```
Hydrogen 2.5.0 
```

Same exact thing! Mine.

[Enter steps to reproduce:]

1. ...
2. ...

**Atom**: 1.27.2 x64
**Electron**: 1.7.15
**OS**: Mac OS X 10.13.4
**Thrown From**: [Hydrogen](ssh://git@github.com/nteract/hydrogen) package 2.5.0


### Stack Trace

Uncaught Error: Cannot find module 'react-hot-loader'

```
At module.js:472

Error: Cannot find module 'react-hot-loader'
    at Module._resolveFilename (module.js:470:15)
    at Module._resolveFilename (/Applications/Atom.app/Contents/Resources/electron.asar/common/reset-search-paths.js:35:12)
    at Function.get_Module._resolveFilename (/Applications/Atom.app/Contents/Resources/app/src/module-cache.js:354:58)
    at Module.require (/app.asar/static/index.js:40:43)
    at require (/Applications/Atom.app/Contents/Resources/app/src/native-compile-cache.js:66:33)
    at /packages/Hydrogen/node_modules/@nteract/transform-vdom/lib/object-to-react.js:13:21
    at /packages/Hydrogen/node_modules/@nteract/transform-vdom/lib/object-to-react.js:16:3)
    at /packages/Hydrogen/node_modules/@nteract/transform-vdom/lib/object-to-react.js:163:3)
    at Module.get_Module._compile (/Applications/Atom.app/Contents/Resources/app/src/native-compile-cache.js:106:36)
    at Object.value [as .js] (/Applications/Atom.app/Contents/Resources/app/src/compile-cache.js:240:29)
    at Module.load (module.js:488:32)
    at tryModuleLoad (module.js:447:12)
    at Function.Module._load (module.js:439:3)
    at Module.require (/app.asar/static/index.js:47:45)
    at require (/Applications/Atom.app/Contents/Resources/app/src/native-compile-cache.js:66:33)
    at /packages/Hydrogen/node_modules/@nteract/transform-vdom/lib/index.js:14:22)
    at /packages/Hydrogen/node_modules/@nteract/transform-vdom/lib/index.js:114:3)
    at Module.get_Module._compile (/Applications/Atom.app/Contents/Resources/app/src/native-compile-cache.js:106:36)
    at Object.value [as .js] (/Applications/Atom.app/Contents/Resources/app/src/compile-cache.js:240:29)
    at Module.load (module.js:488:32)
    at tryModuleLoad (module.js:447:12)
    at Function.Module._load (module.js:439:3)
    at Module.require (/app.asar/static/index.js:47:45)
    at require (/Applications/Atom.app/Contents/Resources/app/src/native-compile-cache.js:66:33)
    at /packages/Hydrogen/node_modules/@nteract/transforms/lib/index.js:43:22)
    at /packages/Hydrogen/node_modules/@nteract/transforms/lib/index.js:110:3)
    at Module.get_Module._compile (/Applications/Atom.app/Contents/Resources/app/src/native-compile-cache.js:106:36)
    at Object.value [as .js] (/Applications/Atom.app/Contents/Resources/app/src/compile-cache.js:240:29)
    at Module.load (module.js:488:32)
    at tryModuleLoad (module.js:447:12)
```

### Commands

```
  2x -1:37 hydrogen:run-and-move-down (input.hidden-input)
     -1:32.3.0 core:move-left (input.hidden-input)
     -1:31.9.0 hydrogen:run-and-move-down (input.hidden-input)
     -1:25.8.0 core:copy (atom-notification.fatal.icon.icon-bug.native-key-bindings.has-detail.has-close.has-stack)
```

### Non-Core Packages

```
autocomplete-python 1.10.5 
busy-signal 1.4.3 
Hydrogen 2.5.0 
intentions 1.1.5 
linter 2.2.0 
linter-ui-default 1.7.1 
pretty-json 2.0.1 
python-indent 1.1.5 
python-tools 0.6.9 
```

[Enter steps to reproduce:]

1. Downloaded AtomSetup-x64.exe v1.27.2
2. Installed Hydrogen 2.5.0 from within Atom
3. Got the error
4. Deleted Hydrogen
5. Installed it with apm
6. Got the same error

**Atom**: 1.27.2 x64
**Electron**: 1.7.15
**OS**: Windows 10
**Thrown From**: [Hydrogen](ssh://git@github.com/nteract/hydrogen) package 2.5.0


### Stack Trace

Uncaught Error: Cannot find module 'react-hot-loader'

```
At module.js:472

Error: Cannot find module 'react-hot-loader'
    at Module._resolveFilename (module.js:470:15)
    at Module._resolveFilename (~/AppData/Local/atom/app-1.27.2/resources/electron.asar/common/reset-search-paths.js:35:12)
    at Function.get_Module._resolveFilename (~/AppData/Local/atom/app-1.27.2/resources/app/src/module-cache.js:354:58)
    at Module.require (/app.asar/static/index.js:40:43)
    at require (~/AppData/Local/atom/app-1.27.2/resources/app/src/native-compile-cache.js:66:33)
    at /packages/Hydrogen/node_modules/@nteract/transform-vdom/lib/object-to-react.js:13:21
    at /packages/Hydrogen/node_modules/@nteract/transform-vdom/lib/object-to-react.js:16:3)
    at /packages/Hydrogen/node_modules/@nteract/transform-vdom/lib/object-to-react.js:163:3)
    at Module.get_Module._compile (~/AppData/Local/atom/app-1.27.2/resources/app/src/native-compile-cache.js:106:36)
    at Object.value [as .js] (~/AppData/Local/atom/app-1.27.2/resources/app/src/compile-cache.js:240:29)
    at Module.load (module.js:488:32)
    at tryModuleLoad (module.js:447:12)
    at Function.Module._load (module.js:439:3)
    at Module.require (/app.asar/static/index.js:47:45)
    at require (~/AppData/Local/atom/app-1.27.2/resources/app/src/native-compile-cache.js:66:33)
    at /packages/Hydrogen/node_modules/@nteract/transform-vdom/lib/index.js:14:22)
    at /packages/Hydrogen/node_modules/@nteract/transform-vdom/lib/index.js:114:3)
    at Module.get_Module._compile (~/AppData/Local/atom/app-1.27.2/resources/app/src/native-compile-cache.js:106:36)
    at Object.value [as .js] (~/AppData/Local/atom/app-1.27.2/resources/app/src/compile-cache.js:240:29)
    at Module.load (module.js:488:32)
    at tryModuleLoad (module.js:447:12)
    at Function.Module._load (module.js:439:3)
    at Module.require (/app.asar/static/index.js:47:45)
    at require (~/AppData/Local/atom/app-1.27.2/resources/app/src/native-compile-cache.js:66:33)
    at /packages/Hydrogen/node_modules/@nteract/transforms/lib/index.js:43:22)
    at /packages/Hydrogen/node_modules/@nteract/transforms/lib/index.js:110:3)
    at Module.get_Module._compile (~/AppData/Local/atom/app-1.27.2/resources/app/src/native-compile-cache.js:106:36)
    at Object.value [as .js] (~/AppData/Local/atom/app-1.27.2/resources/app/src/compile-cache.js:240:29)
    at Module.load (module.js:488:32)
    at tryModuleLoad (module.js:447:12)
```

### Commands

```
```

### Non-Core Packages

```
Hydrogen 2.5.0 
```

i `cd`'ed into `~/.atom/packages/Hydrogen/` and manually installed `react-hot-loader` with `npm install react-hot-loader`, and now get a different error:

```internal/modules/cjs/loader.js:596 throw err; ^

Error: Cannot find module '../build/Release/zmq.node' at Function.Module._resolveFilename (internal/modules/cjs/loader.js:594:15) at Function.Module._load (internal/modules/cjs/loader.js:520:25) at Module.require (internal/modules/cjs/loader.js:650:17) at require (internal/modules/cjs/helpers.js:20:18) at Object. (/usr/local/lib/node_modules/ijavascript/node_modules/zeromq/lib/index.js:6:11) at Module._compile (internal/modules/cjs/loader.js:702:30) at Object.Module._extensions..js (internal/modules/cjs/loader.js:713:10) at Module.load (internal/modules/cjs/loader.js:612:32) at tryModuleLoad (internal/modules/cjs/loader.js:551:12) at Function.Module._load (internal/modules/cjs/loader.js:543:3)
```

now installing zeromq manually let's see what happens

@BenRussert thanks for getting back to me! I rm rf'ed the whole hydrogen folder and then `apm install hydrogen` and now it goes back to giving me this again. (same stack trace as before)

```
Failed to activate the Hydrogen package
Cannot find module 'react-hot-loader'
```

i've also deleted all my atom files and re-installed atom all together. let me know if you have other ideas, or a place for me to look. happy to dig into some code to get this shit working

after i `npm install react-hot-loader` in my `~/.atom/packages/Hydrogen/` directory, i get this error:

```
Javascript (Node.js)
internal/modules/cjs/loader.js:596 throw err; ^

Error: Cannot find module '../build/Release/zmq.node' at Function.Module._resolveFilename (internal/modules/cjs/loader.js:594:15) at Function.Module._load (internal/modules/cjs/loader.js:520:25) at Module.require (internal/modules/cjs/loader.js:650:17) at require (internal/modules/cjs/helpers.js:20:18) at Object. (/usr/local/lib/node_modules/ijavascript/node_modules/zeromq/lib/index.js:6:11) at Module._compile (internal/modules/cjs/loader.js:702:30) at Object.Module._extensions..js (internal/modules/cjs/loader.js:713:10) at Module.load (internal/modules/cjs/loader.js:612:32) at tryModuleLoad (internal/modules/cjs/loader.js:551:12) at Function.Module._load (internal/modules/cjs/loader.js:543:3)
```

before getting this error, hydrogen tells me "kernels updated" and i see all 3 kernels: javascript/node.js, python2 and python3. which is good! but the zeromq stuff is puzzling

--- a/package.json
+++ b/package.json
@@ -68,6 +68,7 @@
     "mobx-react": "^5.0.0",
     "react": "^16.2.0",
     "react-dom": "^16.2.0",
+    "react-hot-loader": "^4.3.3",
     "react-rangeslider": "^2.1.0",
     "spawnteract": "^4.0.0",
     "tildify": "^1.2.0",

